### PR TITLE
Csvt32745 hand in hw3

### PR DESCRIPTION
**Pull request by csvt32745**
This PR including 
- _matrix.cpp: Implementation of matrix class
- matrix_test.py: Unit test module
- MakeFile: Building shared object and running test module
- performance.txt: my benchmark of matrix module

---
A good PR should include the following parts:

- [x] A clear title
- [x] A meaningful message for PR, as well as its commits
- [x] from your specific branch (***not master***) merging to teacher's `HW` branch (***not master***, again)
- [x] excluding any irrelevant files, such as binaries, text files, or dot files
- [x] passing tests/CI

### CODE REVIEW

**Comment by tigercosmos** [comment]
I like your PR message.

Your filename is weird, why `_matrix`?
I thought you were thinking that we don't actually publish that `_matrix`, but in this case, it is suggested that the `Matrix` is public, and the homework is writing a test for it.
Anyway, it is still Okay.

You have commented some code, in a formal project we don't do that

**Comment by tigercosmos** [comment]
just for remind me: get 5

**Comment by tigercosmos** [comment]
just for reminding me: get 5

**Comment by csvt32745** [comment]
> Your filename is weird, why `_matrix`?
> I thought you were thinking that we don't actually publish that `_matrix`, but in this case, it is suggested that the `Matrix` is public, and the homework is writing a test for it.

Yeah I just unconsciously named it as the Python module name

> You have commented some code, in a formal project we don't do that

Got it, thanks :)

File: hw3/csvt32745/matrix_test.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+        self.assertEqual(a.nrow, a.ncol)
+        self.assertEqual(a.nrow, b.nrow)
+        self.assertEqual(a.nrow, c.nrow)
+        self.assertEqual(a.ncol, c.ncol)
+        self.assertEqual(a, c)
+        self.assertNotEqual(id(a), id(c))
+
+    # def testTime(self):
+    #     m = 1234
```
Comment (tigercosmos) [inline hw3/csvt32745/matrix_test.py:43]: remember to remove useless code when you send a PR.

File: hw3/csvt32745/matrix_test.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        c = Matrix(a)
+
+        self.assertEqual(a.nrow, a.ncol)
+        self.assertEqual(a.nrow, b.nrow)
+        self.assertEqual(a.nrow, c.nrow)
+        self.assertEqual(a.ncol, c.ncol)
+        self.assertEqual(a, c)
+        self.assertNotEqual(id(a), id(c))
+
+    # def testTime(self):
```
Comment (tigercosmos) [inline hw3/csvt32745/matrix_test.py:42]: you should let CI creates the performance output.

**Pull request merged**